### ISSUE TITLE
Support nested placeholders

### ISSUE THREAD

**Issue by dafky2000** [dafky2000/simplectemplate#22]
So we can do something like:

```html
{{data.link1:<a href="{{data.link1}}">Test link</a>}}
```
And the link is only displayed if the variable is defined.

**Comment by dafky2000** [comment]
Requires #23

**Comment by dafky2000** [comment]
As per  https://github.com/dafky2000/simplectemplate/issues/23#issuecomment-337225752

For example we should be able to replace:
```
data.key  = "has value"
data.key2 = "also has value"
data.key3 = "maybe"

  {{data.key text to use if the {{data.key2 inner key {{data.key3}}}} has content}}
= {{data.key text to use if the {{data.key2 inner key maybe}} has content}}
= {{data.key text to use if the inner key maybe has content}}
= text to use if the inner key maybe has content
```

**Comment by dafky2000** [comment]
Closed via #61
